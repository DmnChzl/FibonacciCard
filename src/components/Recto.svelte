<script>
	import { beforeUpdate } from 'svelte';
	import Fibonacci from "./Fibonacci.svelte";
  import { isLight } from '../utils';

  export let color = '#212121';
  let background = '#fafafa';

  beforeUpdate(() => {
    if (isLight(color)) {
      background = '#212121';
    }
  });
</script>

<div class="recto" style={`background:${background};`}>
  <Fibonacci color={color} style="top:-1px;left:-1px;" />
  <p style={`color:${color};`}>?</p>
  <Fibonacci color={color} style="bottom:-1px;right:-1px;transform:scaleX(-1)scaleY(-1);" />
</div>

<style>
div {
	position: absolute;
  width: 100%;
  height: 100%;
	border-radius: 0.5rem;
	display: flex;
	justify-content: center;
	align-items: center;
	font-size: 3rem;
  backface-visibility: hidden;
  -webkit-backface-visibility: hidden;
}
</style>
