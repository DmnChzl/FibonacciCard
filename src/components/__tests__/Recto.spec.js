import { render } from '@testing-library/svelte';
import Recto from '../Recto.svelte';

describe('<Recto />', () => {
  it('Testing Snapshot', () => {
    const { container } = render(Recto);

    expect(container).toMatchSnapshot();
  });

  it("Testing 'Color' Prop", () => {
    const { container, queryByText, getByText } = render(Recto, { color: '#FAFAFA' });

    expect(queryByText('?')).toBeInTheDocument();

    const recto = container.querySelector('.recto');
    const text = getByText('?');

    expect(recto).toHaveStyle({ background: '#212121' });
    expect(text).toHaveStyle({ color: '#FAFAFA' });
  });
});
