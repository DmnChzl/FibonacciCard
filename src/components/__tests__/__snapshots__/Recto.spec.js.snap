// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Recto /> Testing Snapshot 1`] = `
<body>
  <div>
    <div
      class="recto svelte-dzf4lg"
      style="background:#FAFAFA;"
    >
      <svg
        fill="none"
        height="146"
        style="position:absolute;top:-1px;left:-1px;"
        viewBox="0 0 102 146"
        width="102"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M100.5 1c0 15.188-12.312 27.5-27.5 27.5-9.389 0-17-7.611-17-17C56 5.701 60.701 1 66.5 1A6.5 6.5 0 0173 7.5a4 4 0 01-4 4A2.5 2.5 0 0166.5 9 1.5 1.5 0 0168 7.5a1 1 0 011 1 .5.5 0 01-1 0M1 56c30.376 0 55 24.624 55 55 0 18.778-15.222 34-34 34-11.598 0-21-9.402-21-21 0-7.18 5.82-13 13-13a8 8 0 018 8 5 5 0 01-5 5 3 3 0 01-3-3 2 2 0 012-2 1 1 0 010 2"
          stroke="#212121"
          stroke-width="2"
        />
      </svg>
       
      <p
        style="color:#212121;"
      >
        ?
      </p>
       
      <svg
        fill="none"
        height="146"
        style="position:absolute;bottom:-1px;right:-1px;transform:scaleX(-1)scaleY(-1);"
        viewBox="0 0 102 146"
        width="102"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M100.5 1c0 15.188-12.312 27.5-27.5 27.5-9.389 0-17-7.611-17-17C56 5.701 60.701 1 66.5 1A6.5 6.5 0 0173 7.5a4 4 0 01-4 4A2.5 2.5 0 0166.5 9 1.5 1.5 0 0168 7.5a1 1 0 011 1 .5.5 0 01-1 0M1 56c30.376 0 55 24.624 55 55 0 18.778-15.222 34-34 34-11.598 0-21-9.402-21-21 0-7.18 5.82-13 13-13a8 8 0 018 8 5 5 0 01-5 5 3 3 0 01-3-3 2 2 0 012-2 1 1 0 010 2"
          stroke="#212121"
          stroke-width="2"
        />
      </svg>
    </div>
  </div>
</body>
`;
